### PR TITLE
Migrate tests to use fluent assertions

### PR DESCRIPTION
**Pull request by jero2rome**
Migrate scroll algorithm tests to use improved assertions using fluent assertions package. If update is accepted, will do migration for all other tests.

### CODE REVIEW

**Comment by bdach** [comment]
I'm not sure why this is being PRed out of the blue without prior discussion. On a quick skim, there are several immediate concerns:

- `FluentAssertions` is an established library, but are we sure we want it over something like https://github.com/ppy/osu-framework/pull/3887?
- If we do decide to adopt the library, it should be done with higher class support. We often write tests using `AddAssertStep()`. That method should ideally be able to naturally be used with fluent assertions.
- Is the library even going to work well with the way we write tests, which heavily relies on delayed execution via delegates? Read [this review thread](https://github.com/ppy/osu-framework/pull/3887#discussion_r489376164) for more info.

Basically this PR is the best case scenario for fluent assertions. I'd like to see what the worst case is before making a decision on whether this should be adopted.

**Comment by bdach** [comment]
[This discord conversation](https://discord.com/channels/188630481301012481/589331078574112768/850331875377545256) is also relevant to this topic (I even proposed fluentassertions adoption there).
